### ISSUE TITLE
Setup environment and add sample data for testing

### ISSUE THREAD

**Issue by Lillingworth1** [Lillingworth1/test_app#5]
Labels: infrastructure, P0-critical, data
Description:
Cannot run the models because:

Dependencies not installed (pandas, scikit-learn, xgboost)
Titanic dataset missing from data/ directory
Need setup instructions
Tasks:

Add instructions for environment setup
Provide sample Titanic dataset or download script
Document how to run each module

**Comment by Lillingworth1** [comment]
✅ Issue #5 completed

Environment setup complete:

Poetry configuration added with all dependencies
Titanic dataset downloaded to data/titanic.csv
Instructions added for installation and running
Commands to use:


poetry install --no-root
poetry run python src/models.py
Commit: 7b31a77
